### PR TITLE
feat(network): add moonbeam network

### LINKED ISSUE #5827: [Bounty] Integrate Beamswap, A DEX Providing Liquidity and Peer to Peer Transactions, into Mask Plugin
## Catalogue

- Platform
  - [x] Gitcoin
  - [ ] Hackerlink
  - [ ] Mask
- Experience
  - [ ] Beginner
  - [x] Intermediate
  - [ ] Advanced
- Category
  - [ ] Mask Core Feature
  - [x] Mask Plugin
  - [ ] Security/Cryptography Feature
  - [ ] Documentation
  - [ ] Design
  - [ ] Other
    - [ ] /_List it here_/
- Time Commitment
  - [x] 2 weeks 
- Bounty Size
  - [x] 1000 USDC 

## Part 1: Problem Description
**Background**  
Beamswap is a decentralized exchange (DEX) with an automated market maker (AMM), providing liquidity and peer-to-peer transactions. But it aims to be so much more than that. Supporting an array of services and features, it will allow you to swap crypto assets, both fungible and non-fungible, earn passive income from staking and yield farming, and even launch your own crypto projects on Moonbeam.

**Solution** Create a plugin that integrates BeamSwap into Mask. The UI should be identical to other swaps.

image

image

## Part 2: Resources

- [Bounty Guide](https://github.com/DimensionDev/Maskbook/tree/master/docs/bounty-guide.md)
- https://beamswap.io/
- https://docs.beamswap.io/
- https://github.com/BeamSwap

## Part 3: Contact and Support
@yisiliu @tzhan28 @LindaMask for ideas
@Jack-Works for implementations
Mask Network discord server: https://discord.gg/4Xccj7a75Q,
Beamswap Discord: https://discord.com/invite/XRpPFHqJ

Please speak up in the Mask discord's bounty section or just discord dm @yisiliu or @taylor zhang or @lin_dao #5827 so that we can find you on Discord. For faster communication, reach out to lin_DAO on telegram, and she will create a dedicated group with the Mask team to start the bounty process.

Submission Requirements
Please include a demo to show the integration is working.

Judging Criteria
Have a working demo.
Clean code & tech design
Clear UI, easy to use and generally good user experience

### PR DESCRIPTION
**Pull request by developerfred**
## Description
Closes #5827 #5826 #5837

Integrate Ecosystem and Network Moonbem

### Exchanges 
- BeamSwap. ✅ 
- Stella Swap  ✅ 
- PadSwap  ✅ 

### Maskbook core contracts 

- [ ] redPacket https://github.com/DimensionDev/MysteryBox/issues/13
- [ ] Multicall 
- [ ] BalanceChecker 
- [ ] MysteryBox https://github.com/DimensionDev/MysteryBox/issues/13
- [ ] ITO - https://github.com/DimensionDev/InitialTwitterOffering/issues/40

### Assets 

- [x] Moonbeam tokenlist https://github.com/DimensionDev/Mask-Token-Lists/issues/18
- [ ]  Moonbeam Assets List

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

**Comment by guanbinrui** [comment]
Hi @developerfred, Plz resolve those conflicts.

**Comment by guanbinrui** [comment]
Hi @developerfred We have deployed contracts on MoonRiver.

Balance Checker: https://moonriver.moonscan.io/address/0xC119574D5Fb333F5AC018658D4d8b5035E16bf39
UniswapInterfaceMulticall: https://moonriver.moonscan.io/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B
ITO: https://moonriver.moonscan.io/address/0x981be454a930479d92C91a0092D204b64845A5D6 at block number: 1314566 
Dummy Qualification Contract：https://[moonriver.moonscan.io/address/0x83D6b366f21e413f214EB077D5378478e71a5eD2](http://moonriver.moonscan.io/address/0x83D6b366f21e413f214EB077D5378478e71a5eD2)
HappyRedPacket: https://moonriver.moonscan.io/address/0x578a7Fee5f0D8CEc7d00578Bf37374C5b95C4b98
HappyRedPacket_ERC721: https://moonriver.moonscan.io/address/0xF9F7C1496c21bC0180f4B64daBE0754ebFc8A8c0

**Comment by developerfred** [comment]
> 90006f9



> Hi @developerfred We have deployed contracts on MoonRiver.
> 
> Balance Checker: [moonriver.moonscan.io/address/0xC119574D5Fb333F5AC018658D4d8b5035E16bf39](https://moonriver.moonscan.io/address/0xC119574D5Fb333F5AC018658D4d8b5035E16bf39) UniswapInterfaceMulticall: [moonriver.moonscan.io/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B](https://moonriver.moonscan.io/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B) ITO: [moonriver.moonscan.io/address/0x981be454a930479d92C91a0092D204b64845A5D6](https://moonriver.moonscan.io/address/0x981be454a930479d92C91a0092D204b64845A5D6) at block number: 1314566 Dummy Qualification Contract：https://[moonriver.moonscan.io/address/0x83D6b366f21e413f214EB077D5378478e71a5eD2](http://moonriver.moonscan.io/address/0x83D6b366f21e413f214EB077D5378478e71a5eD2) HappyRedPacket: [moonriver.moonscan.io/address/0x578a7Fee5f0D8CEc7d00578Bf37374C5b95C4b98](https://moonriver.moonscan.io/address/0x578a7Fee5f0D8CEc7d00578Bf37374C5b95C4b98) HappyRedPacket_ERC721: [moonriver.moonscan.io/address/0xF9F7C1496c21bC0180f4B64daBE0754ebFc8A8c0](https://moonriver.moonscan.io/address/0xF9F7C1496c21bC0180f4B64daBE0754ebFc8A8c0)

done ✅ [90006f9](https://github.com/DimensionDev/Maskbook/pull/5842/commits/90006f95f326cf494edb71c0f6b00762f99a9eec)

File: packages/web3-constants/evm/explorer.json [hunk 1]
```diff
@@ -39,6 +40,7 @@
         "Fantom": "https://api.ftmscan.com/api",
         "Aurora": "https://explorer.mainnet.aurora.dev/api",
         "Aurora_Testnet": "https://explorer.testnet.aurora.dev/api",
-        "Conflux": ""
+        "Conflux": "",
+        "Moonbeam": "https://api-moonbeam.moonscan.io/api?"
```
Comment (guanbinrui) [inline packages/web3-constants/evm/explorer.json]:
```suggestion
        "Moonbeam": "https://api-moonbeam.moonscan.io/api"
```

**Comment by guanbinrui** [comment]
Hi @developerfred Plz fix those CI errors.

**Comment by developerfred** [comment]
@wenluomask can you check this issue for me?

**Comment by developerfred** [comment]
@guanbinrui @Jack-Works 
``bug gas-api.metaswap ``
https://gas-api.metaswap.codefi.network/networks/1284/suggestedGasFees
Screenshot 2022-04-22 at 20 41 29

File: packages/web3-shared/evm/assets/chains.json [hunk 2]
```diff
@@ -1609,14 +1609,14 @@
     },
     {
         "name": "Moonbeam Polkadot",
-        "chain": "MOON",
-        "network": "moonbeam",
-        "rpc": [],
+        "chain": "Moonbeam",
```
Comment (guanbinrui) [inline packages/web3-shared/evm/assets/chains.json]:
```suggestion
        "chain": "moonbeam",
```

File: docs/evm-integration.md [hunk 3]
```diff
@@ -113,9 +113,10 @@ Here is the chain abilities checklist. If you are working on a bounty task, plea
...
 
 | Chain     | Pull Request Link                                    |
-| --------- | ---------------------------------------------------- |
+| --------- | ---------------------------------------------------- | --- |
 | CELO      | <https://github.com/DimensionDev/Maskbook/pull/5052> |
 | Fantom    | <https://github.com/DimensionDev/Maskbook/pull/5036> |
 | xDai      | <https://github.com/DimensionDev/Maskbook/pull/4140> |
 | Arbitrum  | <https://github.com/DimensionDev/Maskbook/pull/3558> |
 | Avalanche | <https://github.com/DimensionDev/Maskbook/pull/5289> |
+| Moonbeam  | <https://github.com/DimensionDev/Maskbook/pull/5842> |     |
```
Comment (guanbinrui) [inline docs/evm-integration.md]:
```suggestion
| --------- | ---------------------------------------------------- |
| CELO      | <https://github.com/DimensionDev/Maskbook/pull/5052> |
| Fantom    | <https://github.com/DimensionDev/Maskbook/pull/5036> |
| xDai      | <https://github.com/DimensionDev/Maskbook/pull/4140> |
| Arbitrum  | <https://github.com/DimensionDev/Maskbook/pull/3558> |
| Avalanche | <https://github.com/DimensionDev/Maskbook/pull/5289> |
| Moonbeam  | <https://github.com/DimensionDev/Maskbook/pull/5842> |
```

**Review by guanbinrui — DISMISSED** [review]

**Comment by wenluomask** [comment]
Currently Moonbeam network is not supported yet. Token list is empty as well.
Could you go through this pr again?
截屏2022-05-02 下午5 26 34
截屏2022-05-02 下午5 26 49

**Comment by developerfred** [comment]
> Currently Moonbeam network is not supported yet. Token list is empty as well.
> Could you go through this pr again

Yes sure

**Comment by wenluomask** [comment]
1.If I use Mask wallet, transactions didn't provide Max priority fee and Max fee;
截屏2022-05-03 下午9 17 09
截屏2022-05-03 下午9 18 33

2. Dashboard cannot get assets in Moonbeam Network;
截屏2022-05-03 下午9 18 59

3. US dollar value of token wasn't added;
截屏2022-05-03 下午9 19 06

4. Since no any DEX supports swap in Moonbeam, why do we open swap in Mask?

**Comment by developerfred** [comment]
Screenshot 2022-05-05 at 22 12 28

**Comment by Jack-Works** [comment]
there're too many merge conflicts 👀

**Comment by wenluomask** [comment]
Hi @developerfred ,

I have checked the latest version of pr just now. I found out that #1-4 were not fixed yet. 

For #1, since you are using gas price instead of priority fee and max fee for gas setting, it is incorrect for Moonbeam. Moonbeam supports EIP-1559, so please change back to EIP-1559 transaction gas setting.

For #4, if Moonbeam did support in BeamSwap, Stella Swap and PadSwap as you checked, the token list should not be empty. Could you check again?
Screen Shot 2022-05-23 at 11 52 05 AM

**Pull request closed without merging**